### ISSUE TITLE
Failed to export spans: Connection Refused: localhost/127.0.0.1:9411

### ISSUE THREAD

**Issue by edtbl76** [open-telemetry/opentelemetry-java#5600]
**Describe the bug**
This looks more like a configuration issue than a bug, but I can't find any docs to help me past it. 

I'm migrating away from spring-sleuth to io.micrometer (spring 3 migration) (Specifically 3.1.1)

    implementation("io.micrometer:micrometer-tracing-bridge-otel:1.1.2")
    implementation("io.opentelemetry:opentelemetry-exporter-zipkin:1.27.0")

This is all I've added to my config. 

```yaml
management:
  tracing:
    sampling:
      probability: 1.0
    enabled: true
    propagation:
      type: w3c
  metrics:
    distribution:
      percentiles-histogram:
        greeting: true
        http:
          server:
            requests: true

logging:
  level:
    root: info
  pattern:
    level: "%5p [${spring.application.name:},%X{traceId:-},%X{spanId:-}]"
```

I've got a docker container running zipkin: 
```yaml
  zipkin:
    image: openzipkin/zipkin:2.24.2
    mem_limit: 1024m
    environment:
      - STORAGE_TYPE=mem
    ports:
      - "9411:9411"
```

When I run my application, I can see spans in the logs, but there is an error when the exporter attempts to ship the information to the container: 
```
023-07-06T04:24:46.071Z  WARN [product-composite,,] 1 --- [or-http-epoll-2] i.o.exporter.zipkin.ZipkinSpanExporter   : Failed to export spans
2023-07-06T04:24:46.072325979Z 
2023-07-06T04:24:46.072328910Z org.springframework.web.reactive.function.client.WebClientRequestException: finishConnect(..) failed: Connection refused: localhost/127.0.0.1:9411
2023-07-06T04:24:46.072331078Z 	at org.springframework.web.reactive.function.client.ExchangeFunctions$DefaultExchangeFunction.lambda$wrapException$9(ExchangeFunctions.java:136) ~[spring-webflux-6.0.9.jar:6.0.9]
2023-07-06T04:24:46.072333178Z 	Suppressed: reactor.core.publisher.FluxOnAssembly$OnAssemblyException: 
2023-07-06T04:24:46.072334996Z Error has been observed at the following site(s):
2023-07-06T04:24:46.072360738Z 	*__checkpoint ⇢ Request to POST http://localhost:9411/api/v2/spans [DefaultWebClient]
2023-07-06T04:24:46.072379523Z Original Stack Trace:
2023-07-06T04:24:46.072381280Z 		at org.springframework.web.reactive.function.client.ExchangeFunctions$DefaultExchangeFunction.lambda$wrapException$9(ExchangeFunctions.java:136) ~[spring-webflux-6.0.9.jar:6.0.9]
2023-07-06T04:24:46.072383220Z 		at reactor.core.publisher.MonoErrorSupplied.subscribe(MonoErrorSupplied.java:55) ~[reactor-core-3.5.6.jar:3.5.6]
2023-07-06T04:24:46.072385103Z 		at reactor.core.publisher.Mono.subscribe(Mono.java:4485) ~[reactor-core-3.5.6.jar:3.5.6]
2023-07-06T04:24:46.072386800Z 		at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:103) ~[reactor-core-3.5.6.jar:3.5.6]
2023-07-06T04:24:46.072389064Z 		at reactor.core.publisher.FluxPeek$PeekSubscriber.onError(FluxPeek.java:222) ~[reactor-core-3.5.6.jar:3.5.6]
2023-07-06T04:24:46.072390942Z 		at reactor.core.publisher.FluxPeek$PeekSubscriber.onError(FluxPeek.java:222) ~[reactor-core-3.5.6.jar:3.5.6]
2023-07-06T04:24:46.072392696Z 		at reactor.core.publisher.FluxPeek$PeekSubscriber.onError(FluxPeek.java:222) ~[reactor-core-3.5.6.jar:3.5.6]
2023-07-06T04:24:46.072394426Z 		at reactor.core.publisher.MonoNext$NextSubscriber.onError(MonoNext.java:93) ~[reactor-core-3.5.6.jar:3.5.6]
2023-07-06T04:24:46.072396158Z 		at reactor.core.publisher.MonoFlatMapMany$FlatMapManyMain.onError(MonoFlatMapMany.java:204) ~[reactor-core-3.5.6.jar:3.5.6]
2023-07-06T04:24:46.072397889Z 		at reactor.core.publisher.SerializedSubscriber.onError(SerializedSubscriber.java:124) ~[reactor-core-3.5.6.jar:3.5.6]
2023-07-06T04:24:46.072400126Z 		at reactor.core.publisher.FluxRetryWhen$RetryWhenMainSubscriber.whenError(FluxRetryWhen.java:225) ~[reactor-core-3.5.6.jar:3.5.6]
2023-07-06T04:24:46.072401977Z 		at reactor.core.publisher.FluxRetryWhen$RetryWhenOtherSubscriber.onError(FluxRetryWhen.java:274) ~[reactor-core-3.5.6.jar:3.5.6]
2023-07-06T04:24:46.072403814Z 		at reactor.core.publisher.FluxContextWrite$ContextWriteSubscriber.onError(FluxContextWrite.java:121) ~[reactor-core-3.5.6.jar:3.5.6]
2023-07-06T04:24:46.072407004Z 		at reactor.core.publisher.FluxConcatMapNoPrefetch$FluxConcatMapNoPrefetchSubscriber.maybeOnError(FluxConcatMapNoPrefetch.java:326) ~[reactor-core-3.5.6.jar:3.5.6]
2023-07-06T04:24:46.072408854Z 		at reactor.core.publisher.FluxConcatMapNoPrefetch$FluxConcatMapNoPrefetchSubscriber.onNext(FluxConcatMapNoPrefetch.java:211) ~[reactor-core-3.5.6.jar:3.5.6]
2023-07-06T04:24:46.072410682Z 		at reactor.core.publisher.FluxContextWrite$ContextWriteSubscriber.onNext(FluxContextWrite.java:107) ~[reactor-core-3.5.6.jar:3.5.6]
2023-07-06T04:24:46.072412491Z 		at reactor.core.publisher.SinkManyEmitterProcessor.drain(SinkManyEmitterProcessor.java:471) ~[reactor-core-3.5.6.jar:3.5.6]
2023-07-06T04:24:46.072414276Z 		at reactor.core.publisher.SinkManyEmitterProcessor$EmitterInner.drainParent(SinkManyEmitterProcessor.java:615) ~[reactor-core-3.5.6.jar:3.5.6]
2023-07-06T04:24:46.072416095Z 		at reactor.core.publisher.FluxPublish$PubSubInner.request(FluxPublish.java:602) ~[reactor-core-3.5.6.jar:3.5.6]
2023-07-06T04:24:46.072427347Z 		at reactor.core.publisher.FluxContextWrite$ContextWriteSubscriber.request(FluxContextWrite.java:136) ~[reactor-core-3.5.6.jar:3.5.6]
2023-07-06T04:24:46.072429352Z 		at reactor.core.publisher.FluxConcatMapNoPrefetch$FluxConcatMapNoPrefetchSubscriber.request(FluxConcatMapNoPrefetch.java:336) ~[reactor-core-3.5.6.jar:3.5.6]
2023-07-06T04:24:46.072431191Z 		at reactor.core.publisher.FluxContextWrite$ContextWriteSubscriber.request(FluxContextWrite.java:136) ~[reactor-core-3.5.6.jar:3.5.6]
2023-07-06T04:24:46.072432990Z 		at reactor.core.publisher.Operators$DeferredSubscription.request(Operators.java:1717) ~[reactor-core-3.5.6.jar:3.5.6]
2023-07-06T04:24:46.072434865Z 		at reactor.core.publisher.FluxRetryWhen$RetryWhenMainSubscriber.onError(FluxRetryWhen.java:192) ~[reactor-core-3.5.6.jar:3.5.6]
2023-07-06T04:24:46.072436756Z 		at reactor.core.publisher.MonoCreate$DefaultMonoSink.error(MonoCreate.java:201) ~[reactor-core-3.5.6.jar:3.5.6]
2023-07-06T04:24:46.072438729Z 		at reactor.netty.http.client.HttpClientConnect$MonoHttpConnect$ClientTransportSubscriber.onError(HttpClientConnect.java:311) ~[reactor-netty-http-1.1.7.jar:1.1.7]
2023-07-06T04:24:46.072440723Z 		at reactor.core.publisher.MonoCreate$DefaultMonoSink.error(MonoCreate.java:201) ~[reactor-core-3.5.6.jar:3.5.6]
2023-07-06T04:24:46.072442592Z 		at reactor.netty.resources.DefaultPooledConnectionProvider$DisposableAcquire.onError(DefaultPooledConnectionProvider.java:162) ~[reactor-netty-core-1.1.7.jar:1.1.7]
2023-07-06T04:24:46.072444457Z 		at reactor.netty.internal.shaded.reactor.pool.AbstractPool$Borrower.fail(AbstractPool.java:475) ~[reactor-netty-core-1.1.7.jar:1.1.7]
2023-07-06T04:24:46.072446393Z 		at reactor.netty.internal.shaded.reactor.pool.SimpleDequePool.lambda$drainLoop$9(SimpleDequePool.java:429) ~[reactor-netty-core-1.1.7.jar:1.1.7]
2023-07-06T04:24:46.072448240Z 		at reactor.core.publisher.FluxDoOnEach$DoOnEachSubscriber.onError(FluxDoOnEach.java:186) ~[reactor-core-3.5.6.jar:3.5.6]
2023-07-06T04:24:46.072450014Z 		at reactor.core.publisher.MonoCreate$DefaultMonoSink.error(MonoCreate.java:201) ~[reactor-core-3.5.6.jar:3.5.6]
2023-07-06T04:24:46.072451930Z 		at reactor.netty.resources.DefaultPooledConnectionProvider$PooledConnectionAllocator$PooledConnectionInitializer.onError(DefaultPooledConnectionProvider.java:560) ~[reactor-netty-core-1.1.7.jar:1.1.7]
2023-07-06T04:24:46.072453854Z 		at reactor.core.publisher.MonoFlatMap$FlatMapMain.secondError(MonoFlatMap.java:241) ~[reactor-core-3.5.6.jar:3.5.6]
2023-07-06T04:24:46.072455619Z 		at reactor.core.publisher.MonoFlatMap$FlatMapInner.onError(MonoFlatMap.java:315) ~[reactor-core-3.5.6.jar:3.5.6]
2023-07-06T04:24:46.072457430Z 		at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:106) ~[reactor-core-3.5.6.jar:3.5.6]
2023-07-06T04:24:46.072459384Z 		at reactor.core.publisher.Operators.error(Operators.java:198) ~[reactor-core-3.5.6.jar:3.5.6]
2023-07-06T04:24:46.072461149Z 		at reactor.core.publisher.MonoError.subscribe(MonoError.java:53) ~[reactor-core-3.5.6.jar:3.5.6]
2023-07-06T04:24:46.072465036Z 		at reactor.core.publisher.Mono.subscribe(Mono.java:4485) ~[reactor-core-3.5.6.jar:3.5.6]
2023-07-06T04:24:46.072467567Z 		at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:103) ~[reactor-core-3.5.6.jar:3.5.6]
2023-07-06T04:24:46.072470752Z 		at reactor.netty.transport.TransportConnector$MonoChannelPromise.tryFailure(TransportConnector.java:587) ~[reactor-netty-core-1.1.7.jar:1.1.7]
2023-07-06T04:24:46.072473992Z 		at reactor.netty.transport.TransportConnector$MonoChannelPromise.setFailure(TransportConnector.java:541) ~[reactor-netty-core-1.1.7.jar:1.1.7]
2023-07-06T04:24:46.072477036Z 		at reactor.netty.transport.TransportConnector.lambda$doConnect$7(TransportConnector.java:265) ~[reactor-netty-core-1.1.7.jar:1.1.7]
2023-07-06T04:24:46.072479840Z 		at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:590) ~[netty-common-4.1.92.Final.jar:4.1.92.Final]
2023-07-06T04:24:46.072482973Z 		at io.netty.util.concurrent.DefaultPromise.notifyListeners0(DefaultPromise.java:583) ~[netty-common-4.1.92.Final.jar:4.1.92.Final]
2023-07-06T04:24:46.072485953Z 		at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:559) ~[netty-common-4.1.92.Final.jar:4.1.92.Final]
2023-07-06T04:24:46.072488921Z 		at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:492) ~[netty-common-4.1.92.Final.jar:4.1.92.Final]
2023-07-06T04:24:46.072492453Z 		at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:636) ~[netty-common-4.1.92.Final.jar:4.1.92.Final]
2023-07-06T04:24:46.072495459Z 		at io.netty.util.concurrent.DefaultPromise.setFailure0(DefaultPromise.java:629) ~[netty-common-4.1.92.Final.jar:4.1.92.Final]
2023-07-06T04:24:46.072498431Z 		at io.netty.util.concurrent.DefaultPromise.tryFailure(DefaultPromise.java:118) ~[netty-common-4.1.92.Final.jar:4.1.92.Final]
2023-07-06T04:24:46.072501515Z 		at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.fulfillConnectPromise(AbstractEpollChannel.java:674) ~[netty-transport-classes-epoll-4.1.92.Final.jar:4.1.92.Final]
2023-07-06T04:24:46.072504858Z 		at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.finishConnect(AbstractEpollChannel.java:693) ~[netty-transport-classes-epoll-4.1.92.Final.jar:4.1.92.Final]
2023-07-06T04:24:46.072508012Z 		at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.epollOutReady(AbstractEpollChannel.java:567) ~[netty-transport-classes-epoll-4.1.92.Final.jar:4.1.92.Final]
2023-07-06T04:24:46.072511451Z 		at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:489) ~[netty-transport-classes-epoll-4.1.92.Final.jar:4.1.92.Final]
2023-07-06T04:24:46.072515359Z 		at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:397) ~[netty-transport-classes-epoll-4.1.92.Final.jar:4.1.92.Final]
2023-07-06T04:24:46.072520218Z 		at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[netty-common-4.1.92.Final.jar:4.1.92.Final]
2023-07-06T04:24:46.072523717Z 		at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.92.Final.jar:4.1.92.Final]
2023-07-06T04:24:46.072549192Z 		at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.92.Final.jar:4.1.92.Final]
2023-07-06T04:24:46.072552596Z 		at java.base/java.lang.Thread.run(Thread.java:833) ~[na:na]
2023-07-06T04:24:46.072555609Z Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: finishConnect(..) failed: Connection refused: localhost/127.0.0.1:9411
2023-07-06T04:24:46.072558985Z Caused by: java.net.ConnectException: finishConnect(..) failed: Connection refused
2023-07-06T04:24:46.072562574Z 	at io.netty.channel.unix.Errors.newConnectException0(Errors.java:166) ~[netty-transport-native-unix-common-4.1.92.Final.jar:4.1.92.Final]
2023-07-06T04:24:46.072565964Z 	at io.netty.channel.unix.Errors.handleConnectErrno(Errors.java:131) ~[netty-transport-native-unix-common-4.1.92.Final.jar:4.1.92.Final]
2023-07-06T04:24:46.072570711Z 	at io.netty.channel.unix.Socket.finishConnect(Socket.java:359) ~[netty-transport-native-unix-common-4.1.92.Final.jar:4.1.92.Final]
2023-07-06T04:24:46.072574171Z 	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.doFinishConnect(AbstractEpollChannel.java:710) ~[netty-transport-classes-epoll-4.1.92.Final.jar:4.1.92.Final]
2023-07-06T04:24:46.072577600Z 	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.finishConnect(AbstractEpollChannel.java:687) ~[netty-transport-classes-epoll-4.1.92.Final.jar:4.1.92.Final]
2023-07-06T04:24:46.072580933Z 	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.epollOutReady(AbstractEpollChannel.java:567) ~[netty-transport-classes-epoll-4.1.92.Final.jar:4.1.92.Final]
2023-07-06T04:24:46.072584287Z 	at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:489) ~[netty-transport-classes-epoll-4.1.92.Final.jar:4.1.92.Final]
2023-07-06T04:24:46.072587686Z 	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:397) ~[netty-transport-classes-epoll-4.1.92.Final.jar:4.1.92.Final]
2023-07-06T04:24:46.072591204Z 	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[netty-common-4.1.92.Final.jar:4.1.92.Final]
2023-07-06T04:24:46.072594785Z 	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.92.Final.jar:4.1.92.Final]
2023-07-06T04:24:46.072598117Z 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.92.Final.jar:4.1.92.Final]
2023-07-06T04:24:46.072601531Z 	at java.base/java.lang.Thread.run(Thread.java:833) ~[na:na]
```

An additional note:
- when I try to set the sysprops/envvars to change the zipkin endpoint, they are ignored. No matter what, the default is always attempted. 
- However, the actual default is completely valid. If I send a request directly to the zipkin container via postman or curl, it works fine. 

**Steps to reproduce**
(Details provided above)

**What did you expect to see?**
I expect to see spans properly exported to zipkin w/o the error mentioned above.

**What did you see instead?**
(See error above)

**What version and what artifacts are you using?**
Versions mentioned above

**Environment**
JDK
```
openjdk version "17.0.7" 2023-04-18 LTS
OpenJDK Runtime Environment Corretto-17.0.7.7.1 (build 17.0.7+7-LTS)
OpenJDK 64-Bit Server VM Corretto-17.0.7.7.1 (build 17.0.7+7-LTS, mixed mode, sharing)
```
OS
```
DISTRIB_ID=Pop
DISTRIB_RELEASE=22.04
DISTRIB_CODENAME=jammy
DISTRIB_DESCRIPTION="Pop!_OS 22.04 LTS"
```

**Additional context**
Add any other context about the problem here

I saw another closed thread -> https://github.com/open-telemetry/opentelemetry-java/issues/5517 
This is similar, but I think that person was more interested in customizing the exporter endpoint. I actually don't need that just yet. (Although I did add the Autoconfiguration settings w/o no affect to the ENV or sys properties)

I'm really just trying to understand why the exporter isn't exporting my spans. 

Thanks in advance!

**Comment by mateuszrzeszutek** [comment]
Hey @edtbl76 ,

Is `localhost:9411` visible from the application's running env?
Are you running both the app and zipkin using a docker compose file? If they're in the same docker network, then perhaps you should modify the endpoint to refer to `zipkin:9411`?

**Comment by edtbl76** [comment]
Yes it's visible. Only zipkin is running in a container. In my description
above I mentioned that I can send requests directly to zipkin without an
issue.

**Comment by jack-berg** [comment]
> when I try to set the sysprops/envvars to change the zipkin endpoint, they are ignored. No matter what, the default is always attempted.

> Although I did add the Autoconfiguration settings w/o no affect to the ENV or sys properties

Can you share the code used to configure the SDK?

**Comment by edtbl76** [comment]
Hi Jack,

That was just additional information. All of the configuration is posted in
the ticket.

As mentioned in the ticket, I'm more interested in the connection failure.
The error logs mention that the connection is refused using the default
zipkin endpoint, however I'm able to POST /spans to that endpoint with
curl/postman/insomnia etc.

**Comment by jack-berg** [comment]
> All of the configuration is posted in
the ticket.

I don't see any programmatic configuration of the SDK, or environment variables / system properties that would be used with autoconfigure.

> As mentioned in the ticket, I'm more interested in the connection failure.

Its unlikely we'll be able to offer helpful advice for that issue. The zipkin exporter uses okhttp3 under the covers. If a connection failure occurs, there's probably something strange going on with the local environment. For example, maybe docker containers aren't exposed on 127.0.0.1 on your machine?

**Comment by jack-berg** [comment]
> There is no documentation in io.micrometer that suggested that this was
necessary for the sleuth -> Micrometter migration path.

> I apologize if this is short-tempered, but so far every response has
include suggestions or questions about something that was provided in the
initial ticket.

You're using a combination of micrometer and opentelemetry tooling, and the folks here only have context on the opentelemetry portions. For us, configuring opentelemetry means programmatic construction of `OpenTelemetrySdk`, or using autoconfigure to interpret [environment variables](https://github.com/open-telemetry/opentelemetry-java/tree/main/sdk-extensions/autoconfigure). Micrometer / slueth might be doing some configuration under the covers, but that's opaque to us.

> I'm more than happy to provide any additional information if it helps
resolve the issue.

When I connect the opentelemetry `ZipkinSpanExporter` to a bogus endpoint, I get a stack trace as follows. Notice the `okhttp.*` stacktraces verse the `reactor.*` in your example. Whatever the connection issue is, its not limited to opentelemetry components. If you're willing to post a small reproduction, I can take a closer look. 
```
[OkHttp http://localhost:9411/...] WARN io.opentelemetry.exporter.zipkin.ZipkinSpanExporter - Failed to export spans
java.net.ConnectException: Failed to connect to localhost/[0:0:0:0:0:0:0:1]:9411
	at okhttp3.internal.connection.RealConnection.connectSocket(RealConnection.kt:297)
	at okhttp3.internal.connection.RealConnection.connect(RealConnection.kt:207)
	at okhttp3.internal.connection.ExchangeFinder.findConnection(ExchangeFinder.kt:226)
	at okhttp3.internal.connection.ExchangeFinder.findHealthyConnection(ExchangeFinder.kt:106)
	at okhttp3.internal.connection.ExchangeFinder.find(ExchangeFinder.kt:74)
	at okhttp3.internal.connection.RealCall.initExchange$okhttp(RealCall.kt:255)
	at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.kt:32)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
	at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.kt:95)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
	at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.kt:83)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
	at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.kt:76)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
	at okhttp3.internal.connection.RealCall.getResponseWithInterceptorChain$okhttp(RealCall.kt:201)
	at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:517)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
	Suppressed: java.net.ConnectException: Failed to connect to localhost/127.0.0.1:9411
		... 19 more
	Caused by: java.net.ConnectException: Connection refused
```

**Comment by edtbl76** [comment]
Interesting. 

I suspect that there is no magic happening under the hood. Is there an example of a "default lightweight" SDK configuration?  Does this have any impact on the connections? 

I'm definitely happy to post a repro. What did you have in mind?

**Comment by jack-berg** [comment]
> Is there an example of a "default lightweight" SDK configuration? Does this have any impact on the connections?

Default for zipkin might look like:
```
    OpenTelemetrySdk openTelemetrySdk =
        OpenTelemetrySdk.builder()
            .setTracerProvider(
                SdkTracerProvider.builder()
                    .addSpanProcessor(
                        BatchSpanProcessor.builder(
                                ZipkinSpanExporter.builder()
                                    .setEndpoint("http://zipkinhost:9411/api/v2/spans")
                                    .build())
                            .build())
                    .build())
            .build();
```
It should not have impact on connections. Nothing we're talking about should impact connections, which is what makes this strange.

> I'm definitely happy to post a repro. What did you have in mind?

A minimal application which contains any dependencies / configuration involved plus a docker compose file with the zipkin service definition.

**Comment by edtbl76** [comment]
Hm. The docker compose file is posted above. I've been using mem storage. (This is the default, so I don't have to define it). All of the configuration is posted in this ticket. The only missing piece is the app itself.  https://github.com/edtbl76/microdaddy/tree/main/product-composite-service <-- this is the module in the app that I've tested out. I see trace/span ids in the logs, it seems like everything is working fine, but it's not getting "out". I'm going to snarf the network to see if I can find something interesting.

**Comment by jack-berg** [comment]
That is strange. For what its worth, I was able to run the application and successfully export spans to zipkin, so it appears that there is some sort of local connectivity issue which is unrelated to code in this project.

<img width="1985" alt="Screenshot 2023-07-07 at 7 05 49 AM" src="https://github.com/open-telemetry/opentelemetry-java/assets/34418638/45940ced-3c8a-43c3-a62a-f91542f84cb0">

**Comment by edtbl76** [comment]
"it worked on my machine!" :)

I found that I was able to get this working after I set
management.zipkin.tracing.endpoint: HTTP://<host>:<port>/api/v2/spans
in application.yml.

However, I think I missed something earlier. I'm able to get this working
outside the IDE, so it seems that it only fails in Intellij.

(FWIW, I created a smaller test app w/ just OTEL and Zipkin, outside of
Spring and Micrometer. I had no issues whatsoever.)

I appreciate the help. We can close this is PEBKAC! :)
